corehq/apps/domain/models.py

```diff
@@ -48,6 +49,19 @@
         lang_lookup[lang['two']] = lang['names'][0]
 
 
+class _StrictQuickCache(QuickCache):
```

Comment:
Great point about still caching, will update that.

Definitely think this is more generalizable but out of scope of this PR. Also going to change the name of `strict` to something better.